src/writer/to_code.rs

```diff
@@ -0,0 +1,31 @@
+use chunk::{Chunk, Chunks};
+use super::defines::Strings;
+use super::primitives::*;
+
+pub trait ToCode {
+    fn to_code(&self) -> Strings;
+}
+
+impl ToCode for Chunk {
+    fn to_code(&self) -> Strings {
+        use chunk::Chunk::*;
+        match *self {
+            Comment(ref chs) => comment_block(chs.to_code()),
+            BlockHalf(ref chs) => format_block("", "}", chs.to_code()),
+            Unsafe(ref chs) => format_block_smart("unsafe {", "}", chs.to_code(), " ", " "),
+            FfiCallTODO(ref name) => vec![format!("TODO: call ffi:{}()", name)],
+        }
+    }
+}
+
+impl ToCode for Chunks {
+    fn to_code(&self) -> Strings {
+        let mut v = Strings::new();
+        for ch in self {
+            let s = ch.to_code();
+            //TODO: append
+            s.iter().map(|s| v.push(s.clone())).count();
```

Comment:
It's not recommended to use `map` for side-effects. A `for` loop here would be more idiomatic.